awx/main/access.py

```diff
@@ -464,9 +464,19 @@ def can_add(self, data):
         return self.user.is_superuser
 
     def can_change(self, obj, data):
+        if obj.name == "tower":
+            return False
         return self.user.is_superuser
 
     def can_delete(self, obj):
+        if obj.name == "tower":
+            return False
```

Comment:
The views have a property `always_allow_superuser` which defaults to True (I believe is the case for `InstanceGroupDetail`). That bakes in the assumption that the access methods only check permissions, and one would expect that to circumvent checks like these.